code-gen-projects/java/code-gen-demo/build.gradle.kts

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * This generated file contains a sample Java library project to get you started.
+ * For more details on building Java & JVM projects, please refer to https://docs.gradle.org/8.6/userguide/building_java_projects.html in the Gradle documentation.
+ */
+
+plugins {
+    // Apply the java plugin for API and implementation separation.
+    java
+}
+
+repositories {
+    // Use Maven Central for resolving dependencies.
+    mavenCentral()
+}
+
+dependencies {
+    implementation("com.amazon.ion:ion-java:1.11.4")
+
+    // Use JUnit Jupiter for testing.
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+    testImplementation("org.junit.jupiter:junit-jupiter:5.7.1")
+}
+
+val ionSchemaSourceCodeDir = "../../schema/"
+val generatedIonSchemaModelDir = "${layout.buildDirectory.get()}/generated/java"
+sourceSets {
+    main {
+        java.srcDir(generatedIonSchemaModelDir)
+    }
+}
+
+
+tasks {
+    val ionCodegen = create<Exec>("ionCodegen") {
+        inputs.files(ionSchemaSourceCodeDir)
+        outputs.file(generatedIonSchemaModelDir)
+
+        val ionCli = "../../../target/debug/ion"
```

Comment:
I think you can pass the target directory using gradle properties. E.g.:

Set the path as an environment variable in the Rust integration test: https://github.com/popematt/ion-cli/blob/a24ecd23d659f6c0142b20bbc8a4771ac6a01c80/tests/codegen-tests.rs#L59-L65

Access the path from the gradle build script: https://github.com/popematt/ion-cli/blob/a24ecd23d659f6c0142b20bbc8a4771ac6a01c80/codegen-test-projects/java/build.gradle.kts#L33

Reply:
Folow-up PR: #98